src/lib/update-lock.ts

```diff
@@ -115,47 +121,62 @@ const lockExistsErrHandler = (err: Error, release: () => void) => {
 /**
  * Try to take the locks for an application. If force is set, it will remove
  * all existing lockfiles before performing the operation
+ *
+ * TODO: convert to native Promises. May require native implementation of Bluebird's dispose / using
  */
-export function lock(
+export function lock<T extends unknown>(
 	appId: number | null,
-	{ force = false }: { force: boolean },
-	fn: () => PromiseLike<void>,
-): Bluebird<void> {
+	{ force = false, skipLock = false }: { force: boolean; skipLock?: boolean },
+	fn: () => Resolvable<T>,
+): Bluebird<T> {
+	if (skipLock) {
```

Comment:
Not a comment on your code, I really hate this `skipLock` stuff. I hope we can remove it one day. I think it's only necessary because of the implementation of those functions with the `pausingApply`.

Reply:
I think this might be my only argument about merging `lockingIfNecessary` into this module. If `update-lock` is though as an independent locking module, I'm not sure it would make sense to have the `skipLock` option.

I don't think this is necessarily a good enough reason to remove this duplication, but at least we should make a note to remove this option in the future. Modules should either use the lock or not use the lock, but having `skipLock` to bypass the lock on some occasions is just a bad API

Reply:
> I don't think this is necessarily a good enough reason to remove this duplication

Could you clarify? I'm taking this to mean, we shouldn't remove lockingIfNecessary in favor of the base lock function because of skipLock. If this is not the case, I can add a TODO comment here to remove skipLock?

Reply:
Oh, yes, I said the opposite of what I meant, which is that we keep these changes but add a `FIXME` comment to remove `skipLock`